execution/evm/vm.go

```diff
@@ -733,21 +733,31 @@ func (vm *VM) execute(callState Interface, eventSink EventSink, caller, callee c
 			}))
 			vm.Debugf(" => T:%X D:%X\n", topics, data)
 
-		case CREATE: // 0xF0
+		case CREATE, CREATE2: // 0xF0, 0xFB
 			returnData = nil
+			var salt Word256
 
 			contractValue := stack.PopU64()
 			offset, size := stack.PopBigInt(), stack.PopBigInt()
 			input := memory.Read(offset, size)
 
+			if op == CREATE2 {
+				salt = stack.Pop()
+			}
+
 			// TODO charge for gas to create account _ the code length * GasCreateByte
 			useGasNegative(gas, GasCreateAccount, callState)
 
 			vm.sequence++
 			nonce := make([]byte, txs.HashLength+uint64Length)
 			copy(nonce, vm.tx.Hash())
 			PutUint64BE(nonce[txs.HashLength:], vm.sequence)
-			newAccount := crypto.NewContractAddress(callee, nonce)
+			var newAccount crypto.Address
+			if op == CREATE {
```

Comment:
Move all the noncense above into the `op == CREATE` branch (hmm including or excluding `vm.sequence++` - can see argument both ways, but it's not necessary to increment given we are aren't using so I thin we probably move it within the `CREATE` branch).

We don't need to do it since we will not be using the nonce.